doc/rst/feat_adat_tx.rst

```diff
@@ -0,0 +1,134 @@
+
+    on stdcore[AUDIO_IO_TILE] : buffered out port:32 p_adat_tx  = PORT_ADAT_OUT;
+
+This port should be clocked from the master-clock::
+
+    configure_out_port_no_ready(p_adat_tx, clk_audio_mclk, 0);
+    set_clock_fall_delay(clk_audio_mclk, 7);
+
+
+Finally the ADAT transmitter task is run - passing in the port and channel for communication with ``XUA_AudioHub``::
+
+    adat_tx_port(c_adat_out, p_adat_tx);
+
+
+.. _xua_adat_tx:
+
+Communication between ``XUA_AudioHub`` and the ``adat_tx_port`` task
+--------------------------------------------------------------------
+
+The interface to the ADAT transmitter core is via a normal channel with streaming builtins (``outuint``, ``inuint``).
+
+To begin with, ``XUA_AudioHub`` sends two values on the channel - the master clock multiplier and
+the S/MUX setting.
+
+The master clock multiplier is the ratio between the mclk freqency and the sampling frequency. The S/MUX setting is
+1, 2 or 4, depending on the sampling frequency:
+
+.. literalinclude:: ../../lib_xua/src/core/audiohub/xua_audiohub.xc
+   :start-at: /* Calculate what master clock we should be using */
+   :end-before: /* Calculate master clock to bit clock
+
+This is followed by communicating the address of a block of memory holding the audio samples block.
+The ``XUA_AudioHub`` "runs ahead" of the ADAT transmitter task, assembling the next sample block while the
+ADAT transmitter converts the current block into an ADAT stream to transmit over the optical interface.
+
+The ADAT transmitter, once done processing the current block, acknowledges this by sending a data token over the channel
+to ``XUA_AudioHub`` as a handshake mechanism.
+On receiving this handshake, ``XUA_AudioHub`` sends the address of the next block of samples over the channel.
+
+Note that a ``XS1_CT_END`` end token is not sent between blocks of data, leading to the channel remaining open and getting used
+as a streaming channel.
+
+``XUA_AudioHub`` only terminates the connection by sending a ``XS1_CT_END`` token when there's a sampling frequency change
+that requires the ``XUA_AudioHub`` to re-communicate the master clock multiplier and
+the S/MUX setting.
+
+In case of a sampling frequency change, the ``XUA_AudioHub`` receives the pending handshake from the ADAT transmitter,
+followed by sending the ``XS1_CT_END`` token indicating the end of data streaming to the ADAT task.
+A fresh transmission is then started by sending the new master clock multiplier and
+the S/MUX setting to the ADAT transmitter, followed by audio blocks transfer as described above.
+
+:ref:`Communication between AudioHub and ADAT Transmitter<xua_audiohub_adat_tx>` describes the communication between ``XUA_AudioHub`` and the ADAT transmitter:
+
+ .. _xua_audiohub_adat_tx:
+
+ .. figure:: images/xua_audiohub_adat_tx.png
+   :width: 60%
+
+For further details please see the documentation, application notes and examples provided for ``lib_adat``.
```

Comment:
would be nice to hyperlink to lib_adat at https://www.xmos.com/file/lib_adat

It's not there yet, but will be soon!

Reply:
done